project-base/tests/ShopBundle/Functional/Model/Cart/CartTest.php

```diff
@@ -56,4 +56,31 @@ public function testRemoveItem()
         $cart->removeItemById($cartItem1->getId());
         $this->assertSame(1, $cart->getItemsCount());
     }
+
+    public function testCleanCartMakesCartEmpty()
+    {
+        $productDataFactory = $this->getContainer()->get(ProductDataFactoryInterface::class);
+
+        $price = 100;
+        $vatData = new VatData();
+        $vatData->name = 'vat';
+        $vatData->percent = 21;
+        $vat = new Vat($vatData);
+
+        $productData = $productDataFactory->create();
+        $productData->name = ['cs' => 'Any name'];
+        $productData->price = $price;
+        $productData->vat = $vat;
+        $product = Product::create($productData);
```

Comment:
Maybe it would be easier to persistence reference instead of creating a product on these 10 lines :slightly_smiling_face:

Reply:
Creating product is changed in next commit to a private method

Reply:
...but never mind :laughing: